pkg/scheduler/core/generic_scheduler.go

```diff
@@ -913,6 +913,10 @@ func filterPodsWithPDBViolation(pods []*v1.Pod, pdbs []*policy.PodDisruptionBudg
 				if selector.Empty() || !selector.Matches(labels.Set(pod.Labels)) {
 					continue
 				}
+				// If the pod exists in DisruptedPods, add the pod into nonViolatingPods directly.
+				if _, exist := pdb.Status.DisruptedPods[pod.Name]; exist {
```

Comment:
It's accurate based on my understanding of `pdb.Status.DisruptedPods` - the pods in this map should have been marked as deleted by API server.

@ahg-g it'd be good if you can take a second look.

Reply:
It seems the [api server's eviction](https://github.com/kubernetes/kubernetes/blob/35fc65dc2c614e57e5a019358a24af78ee1cfc16/pkg/registry/core/pod/storage/eviction.go#L193) doesn't support more than one PodDisruptionBudget per pod, so the ```break``` at link 923 should be moved to after the if/else (i.e., right after line 925):

Reply:
or we can support that just in case like this:

```
// We have found a matching PDB.
if pdbsAllowed[i] <= 0 {
  pdbForPodIsViolated = true
} else {
  if _, exist := pdb.Status.DisruptedPods[pod.Name]; !exist {
    pdbsAllowed[i]--
  }
}
```

Reply:
> but what is concerning is having the pod.Name as the key since it is not unique across namespaces,

I think it's b/c PDB is namespaced resouce, so it's assumed to only match the pods that have the same namespace as itself. It's checked at L905.

Reply:
> I think it's b/c PDB is namespaced resouce, so it's assumed to only match the pods that have the same namespace as itself. It's checked at L905.

ah, right.

Reply:
I find that the eviction will update the pdb.Status.DisruptedPods before deleteing the pod.
https://github.com/kubernetes/kubernetes/blob/15e95e48964d4e800f7210a249f9122ee7d79fcc/pkg/registry/core/pod/storage/eviction.go#L216-L219

And  in `checkAndDecrement`, it not only add the evicted pod in `pdb.Status.DisruptedPods` but also decrement the `pdb.Status.DisruptionsAllowed`. 
https://github.com/kubernetes/kubernetes/blob/15e95e48964d4e800f7210a249f9122ee7d79fcc/pkg/registry/core/pod/storage/eviction.go#L293-L310

So  `pdbsAllowed[i] = pdb.Status.DisruptionsAllowed` is right. We just need to add the pod into nonViolatingPods directly in `filterPodsWithPDBViolation` if the pod exists in DisruptedPods.
@ahg-g  @Huang-Wei